app/QueryManager.py

```diff
@@ -20,12 +20,65 @@ def trips_count(dataframe, date_column):
+                                                                                                    'vendor_id')))
+                         .filter(col('count') == col('max_trip_count')).drop('max_trip_count'))
         return trips_by_week
+
+    def total_revenue(self):
+        """ Calculates the total revenue of each vendor """
+        dataframe = (self.trip_fare_df.filter(col(columns.vendor_id) != 'None').groupBy(columns.vendor_id)
+                     .agg(format_number(sum(columns.total_amount), 2).alias('total revenue')))
+        return dataframe
+
+    def avg_trip_distance(self):
+        """ Calculates the average trip distance for different amount of passengers """
+        dataframe = (self.trip_data_df.filter(col(columns.passenger_count).
+                                              isNotNull()).groupBy(columns.vendor_id, columns.passenger_count).
+                     agg(avg(columns.trip_distance)).orderBy(desc(columns.passenger_count)))
+        return dataframe
+
+    def simultaneous_trips(self):
+        """ Calculates how many trips happened simultaneously """
+        pickup_dataframe = (self.trip_data_df.filter(col(columns.pickup_datetime).isNotNull()).
+                            select(col(columns.pickup_datetime).alias('event_time'),
+                                   lit(1).alias('event_count')))
+        dropoff_dateframe = (self.trip_data_df.filter(col(columns.dropoff_datetime).isNotNull()).
+                             select(col(columns.dropoff_datetime).alias('event_time'),
+                                    lit(-1).alias('event_count')))
+        event_dateframe = pickup_dataframe.union(dropoff_dateframe)
+        dataframe = event_dateframe.withColumn('sum', sum('event_count').over(Window.partitionBy('event_time')
+                                                                              .orderBy(asc('event_time'))))
+        dataframe = dataframe.groupBy(date_format('event_time', 'yyyy-MM-dd').alias('day')
+                                      ).agg(max('sum').alias('simultaneous_trips')).orderBy(
+            desc(col('simultaneous_trips'))).limit(5)
+        return dataframe
+
+    def most_expensive_trips(self):
+        """Calculates the most expensive trips by vendor """
+        dataframe = self.trip_fare_df.groupBy(columns.vendor_id).agg(max(columns.total_amount).
+                                                                     alias(columns.total_amount))
+        return dataframe
+
+    def avg_amount_rate_code(self):
+        """ Calculates count of trips the tip above average tip amount for trips with different rate code. """
+        dataframe = self.trip_fare_df.join(
+                    self.trip_data_df,
+                    [self.trip_data_df[columns.medallion] == self.trip_fare_df[columns.medallion],
+                     self.trip_data_df[columns.hack_license] == self.trip_fare_df[columns.hack_license],
```

Comment:
isn't easier to just pass the list of names of columns instead of checking the condition for each?